app/src/main/java/com/jerboa/ui/components/common/TimeAgo.kt

```diff
@@ -52,25 +63,48 @@ fun TimeAgo(
         )
 
         updated?.also {
-            val updatedPretty = dateStringToPretty(it, longTimeFormat)
-
             DotSpacer(
                 padding = SMALL_PADDING,
                 style = MaterialTheme.typography.bodyMedium,
             )
-            Text(
-                text = "($updatedPretty)",
-                style = MaterialTheme.typography.bodyMedium,
-                color = MaterialTheme.colorScheme.onBackground.muted,
-                fontStyle = FontStyle.Italic,
-            )
+            val updatedPretty = dateStringToPretty(it, longTimeFormat)
+
+            if (updatedPretty == null) {
+                Text(
+                    text = stringResource(R.string.time_ago_failed_to_parse),
+                    color = MaterialTheme.colorScheme.error,
+                    style = MaterialTheme.typography.labelSmall,
+                )
+            } else {
+                Text(
+                    text = "($updatedPretty)",
+                    style = MaterialTheme.typography.bodyMedium,
+                    color = MaterialTheme.colorScheme.onBackground.muted,
+                    fontStyle = FontStyle.Italic,
+                )
+            }
         }
     }
 }
 
-fun dateStringToPretty(dateStr: String, longTimeFormat: Boolean = false): String {
-    val publishedDate = Date.from(Instant.parse(dateStr + "Z"))
-    return formatDuration(publishedDate, longTimeFormat)
+/**
+ * Converts a date string to a pretty string like "2 hours" or "2h"
+ *
+ * @param dateStr The date string to convert
+ * @param longTimeFormat If true, use a long time format like "2 hours, 3 minutes ago"
+ * @return The pretty string, or null if the date string could not be parsed
+ */
+fun dateStringToPretty(dateStr: String, longTimeFormat: Boolean = false): String? {
+    return try {
+        // TODO: Remove this hack once backward API compatibility is implemented
```

Comment:
I'm wondering if we should create some sort of standard for naming these hacks and older versions... or even just the TODO comment. Something like `TODO v0.18.4_deprecated`

Reply:
We could but I don't think we will have many. This is only one right now and I don't foresee any others.